src/app/loadout/ModPicker.tsx

```diff
@@ -117,28 +120,30 @@ function mapStateToProps() {
           (socket) => socket.socketDefinition.reusablePlugSetHash
         );
 
+        // For each of the socket types on the item, figure out what plugs could go into it
+        // and the maximum number of those sockets that can appear on a single item.
         for (const [hashAsString, sockets] of Object.entries(socketsGroupedByPlugSetHash)) {
           const plugSetHash = parseInt(hashAsString, 10);
-          const plugsWithDuplicates: PluggableInventoryItemDefinition[] = [];
-
-          const dimPlugs = filterDimPlugsAvailableToCharacerOrProfile(
+          const dimPlugs = filterDimPlugsUnlockedOnCharacterOrProfile(
             profileResponse,
-            sockets.find((socket) => socket.plugSet)!.plugSet!,
+            sockets[0].plugSet!,
             // TODO: For vaulted items, union all the unlocks and then be smart about picking the right store
             owner ?? currentStore!.id
           );
 
-          // Get the item plugs actually available to the profile
+          // Filter down to plugs that match the plugCategoryHashWhitelist
+          const plugsWithDuplicates: PluggableInventoryItemDefinition[] = [];
           for (const dimPlug of dimPlugs) {
             if (
               isInsertableArmor2Mod(dimPlug.plugDef) &&
               (!plugCategoryHashWhitelist ||
-                plugCategoryHashWhitelist?.includes(dimPlug.plugDef.plug.plugCategoryHash))
+                plugCategoryHashWhitelist.includes(dimPlug.plugDef.plug.plugCategoryHash))
             ) {
               plugsWithDuplicates.push(dimPlug.plugDef);
             }
           }
 
+          // TODO: Why would there be duplicates within a single plugset?
```

Comment:
I think this is hangover from the first pass at fixing artificer mods. I think we can clean this up